sci-libs/gdal/gdal-2.0.3.ebuild

```diff
@@ -0,0 +1,294 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="6"
+
+WANT_AUTOCONF="2.5"
+
+GENTOO_DEPEND_ON_PERL="no"
+PYTHON_COMPAT=( python2_7 python3_{4,5} )
+DISTUTILS_OPTIONAL=1
+
+inherit autotools perl-module distutils-r1 toolchain-funcs java-pkg-opt-2
+
+DESCRIPTION="Translator library for raster geospatial data formats (includes OGR support)"
+HOMEPAGE="http://www.gdal.org/"
+SRC_URI="http://download.osgeo.org/${PN}/${PV}/${P}.tar.gz"
+
+SLOT="0/2"
+LICENSE="BSD Info-ZIP MIT"
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x86-macos"
+IUSE="armadillo +aux_xml curl debug doc fits geos gif gml hdf5 java jpeg jpeg2k mdb mysql netcdf odbc ogdi opencl oracle pdf perl png postgres python spatialite sqlite threads xls"
+
+COMMON_DEPEND="dev-libs/expat
+	dev-libs/json-c:=
+	dev-libs/libpcre
+	dev-libs/libxml2:=
+	media-libs/tiff:0=
+	sci-libs/libgeotiff
+	sys-libs/zlib[minizip(+)]
+	armadillo? ( sci-libs/armadillo:=[lapack] )
+	curl? ( net-misc/curl )
+	fits? ( sci-libs/cfitsio:= )
+	geos?   ( >=sci-libs/geos-2.2.1 )
+	gif? ( media-libs/giflib:= )
+	gml? ( >=dev-libs/xerces-c-3 )
+	hdf5? ( >=sci-libs/hdf5-1.6.4:=[szip] )
+	java? ( >=virtual/jre-1.6:* )
+	jpeg? ( virtual/jpeg:0= )
+	jpeg2k? ( media-libs/openjpeg:2= )
+	mdb? ( dev-java/jackcess )
+	mysql? ( virtual/mysql )
+	netcdf? ( sci-libs/netcdf:= )
+	odbc?   ( dev-db/unixODBC )
+	ogdi? ( sci-libs/ogdi )
+	opencl? ( virtual/opencl )
+	oracle? ( dev-db/oracle-instantclient:= )
+	pdf? ( >=app-text/poppler-0.24.3:= )
+	perl? ( dev-lang/perl:= )
+	png? ( media-libs/libpng:0= )
+	postgres? ( >=dev-db/postgresql-8.4:= )
+	python? (
+		${PYTHON_DEPS}
+		dev-python/setuptools[${PYTHON_USEDEP}]
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+	sqlite? ( dev-db/sqlite:3 )
+	spatialite? ( dev-db/spatialite )
+	xls? ( dev-libs/freexl )"
+
+RDEPEND="${COMMON_DEPEND}"
+
+DEPEND="${COMMON_DEPEND}
+	doc? ( app-doc/doxygen )
+	perl? ( dev-lang/swig:0 )
+	python? ( dev-lang/swig:0 )"
+
+REQUIRED_USE="mdb? ( java )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	spatialite? ( sqlite )
+"
+
+pkg_setup() {
```

Comment:
That's strange. It's exported by java-pkg-opt-2. I think there's a pkg_setup exported by the Python stuff (hard to tell) that you should probably be calling too. It won't be overriding the Java one though as java-pkg-opt-2 is last in the inherit list.

I tried it myself by doing `USE="java mdb" ebuild gdal-2.0.3.ebuild clean setup` and, assuming you have `virtual/jdk` rather than `virtual/jre`, you see the line `Using: icedtea-bin-8` whether `pkg_setup` is present or not. It only goes away if you have a blank `pkg_setup` so it's definitely being called. Something weird with your test?